code/modules/telesci/gps.dm

```diff
@@ -89,19 +79,19 @@ var/list/SPS_list = list()
 	var/data[0]
 	if(emped)
 		data["emped"] = TRUE
-	else if (transmitting)
-		data["gpstag"] = gpstag
-		data["autorefresh"] = autorefreshing
-		data["location_text"] = get_location_name()
-		var/list/devices = list()
-		for(var/D in gps_list)
-			var/obj/item/device/gps/G = D
-			if(G.transmitting && src != G)
-				var/device_data[0]
-				device_data["tag"] = G.gpstag
-				device_data["location_text"] = G.get_location_name()
-				devices += list(device_data)
-		data["devices"] = devices
+	data["transmitting"] = transmitting
+	data["gpstag"] = gpstag
+	data["autorefresh"] = autorefreshing
+	data["location_text"] = get_location_name()
+	var/list/devices = list()
+	for(var/D in gps_list)
```

Comment:
Is there a reason for this to be a typeless loop ?

Reply:
Typeless loops don't typecheck, so they're marginally faster, or as I've recently taken to calling it, TG-faster. Also I didn't change that.

Reply:
Fine by me